indent/handlebars.vim

```diff
@@ -79,16 +79,58 @@ function! GetHandlebarsIndent(...)
   " all indent rules only apply if the block opening/closing
   " tag is on a separate line
 
-  " indent after block {{#block
-  if prevLine =~# '\v\s*\{\{[#^].*\s*'
+  " " check for a hanging attribute
+  let lastLnumCol = col([lnum, '$']) - 1
+  if synIDattr(synID(lnum, lastLnumCol, 1), "name") =~ '^mustache'
+        \ && prevLine !~# '}}\s*$'
+    let hangingAttributePattern = '{{[#^]\=\%(\k\|[/-]\)\+\s\+\zs\k\+='
+    let standaloneComponentPattern = '^\s*{{\%(\k\|[/-]\)\+\s*$'
+
+    if prevLine =~ hangingAttributePattern
+      " {{component attribute=value
+      "             other=value}}
+      let [line, col] = searchpos(hangingAttributePattern, 'Wbn', lnum)
+      if line == lnum
+        return col - 1
+      endif
+    elseif prevLine =~ standaloneComponentPattern
+      " {{component
+      "   attribute=value}}
+      return indent(lnum) + sw
+    endif
+  endif
+
+  let componentClosingPattern = '\v^\s*\{\{\/\S*\}\}\s*'
```

Comment:
Can we move this definition closer to the beginning of the script, and scope it with `s:`?